core/arch/arm/plat-stm32mp1/sub.mk

```diff
@@ -3,6 +3,7 @@ global-incdirs-y += .
 srcs-y += main.c
 srcs-y += reset.S
 srcs-$(CFG_SCMI_MSG_DRIVERS) += scmi_server.c
+srcs-$(CFG_STM32MP1_HUK_BSEC) += plat_huk.c
```

Comment:
As already comment, the file can be better called stm32mp15_huk.c as it will be MP15 specific.

Reply:
Perhaps we could keep the same file name and resolve the implementation internally?

Reply:
we already design a solution that will not use at all this way of working, upstream will come soon for MP13.

Reply:
you will still need to provide `tee_otp_get_hw_unique_key`

Reply:
On stm32mp13, the HUK is not accessible by the CPU. We will proposed on that a weak on the `huk_subkey_derive` function rather. In that case the `tee_otp_get_hw_unique_key` implementation will not be used at all.

Reply:
Seems weird to me to break that interface...so what will that huk interface provide fomr mp13? because it has to do something....
in any case, I see no reason for not merging what I am proposing... you can propose your change later on and do the necessary modifications....I believe this is the way the open source projects usually works.

Reply:
You're right, the future implementation is out of scope of the discussion. My point is that the file named plat_huk.c means usable for all stm32mp1 which is the purpose of my comment because this is not usable on MP13 where the huk is not coming directly from OTP. So, I'd prefer, for more clarity to keep that under specific stm32mp**15** implementation.

Reply:
right but regardless - no matter what smt32mp13 does - OP-TEE must have a` tee_otp_get_hw_unique_key` function which is all that this file provides for stm32mp1

I think it makes sense this way. Then for mp13 we can just add comments - if that is all it is needed - on the file perhaps describing its implementation.

Otherwise this plat-huk.c will have to go into core/drivers/stmp32mp15-huk.c and users will have to dig for the huk information for mp13...it makes it all more difficult to follow IMO.

Reply:
As Lionel mentioned in https://github.com/OP-TEE/optee_os/pull/5555#discussion_r987711168, mp13 will not implement `tee_otp_get_hw_unique_key()` but rather `huk_subkey_derive()` (which needs to be overridable) subce the hw will do the huk derivation, OP-TEE won't manipulate the huk itself. Hence another implementation.

I would suggest renaming s/CFG_STM32MP1_HUK/CFG_STM32MP15_HUK/.

I think our dear maintainers prefer to have drivers in _core/drivers/_ rather than _core/.../plat-xxxx/_.  In that scope _core/drivers/stm32mp15_huk.c_ could be a good candidate.

Reply:
Sorry but no, that is not true: mp13 - as every OP-TEE platform - has to have an implementation of tee_otp_get_hw_unique_key(): it can provide it explicitly or fallback to the weak implementation. But there has to be one implementation.

This PR is just explicit about it. It doesn't preclude changes to huk_subkey_derive()....so I dont see what is wrong with it.

Reply:
I don't agree. For a platform not accessing a huk but only able to perfom derivation, there should be not need (no point) to implement  `tee_otp_get_hw_unique_key()`. In the generic code, `tee_otp_get_hw_unique_key()` is only called from _huk_subkey.c_ in the purpose of generating the derived keys. This where function `huk_subkey_derive()` will need to be overrideable by a platform specific implementation.

Reply:
Let's not anticipate too much here. I tend to agree with @etienne-lms that the proper interface between the generic code and the platform-specific code shall become `huk_subkey_derive()` rather than `tee_otp_get_hw_unique_key()`, but until then let's keep things as simple and clear as possible. Referring to Lionel's comment:

>  As already comment, the file can be better called stm32mp15_huk.c as it will be MP15 specific.

...and Etienne's one:

> I would suggest renaming s/CFG_STM32MP1_HUK/CFG_STM32MP15_HUK/.

...both seem totally reasonable to me. This code is intended for use by MP15 so call it MP15, period. Whether some MP15 code will fall under MP13 or MP1 we can decide later.

Reply:
ok lets do that and keep it simple: let me send my PR in the simplest possible terms and if someone needs additional functionality it can happen later.